### PR TITLE
chore(build): Split dist packaging into its own script

### PR DESCRIPTION
**Pull request by martinjlowm**
This adds convenience in local development in other projects like so:

```
git clone https://github.com/Typescript-TDD/ts-auto-mock.git
git checkout <custom-branch>
npm ci

npm run build
npm run dist:collect
```

or through symlinks to a local copy (ts-auto-mock/dist) with custom changes.

---

Personally, I use Yarn 2 and the `exec`-protocol to execute the script above allowing me to depend on a patched version of ts-auto-mock.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
     "test:features": "cross-env CACHE=true FEATURES=true karma start config/karma/karma.config.features.js",
     "test:playground": "karma start config/karma/karma.config.transformer.playground.js",
     "test:playground:build": "karma start config/karma/karma.config.transformer.playground.build.js",
-    "release": "cp -r package.json package-lock.json README.md dist && semantic-release",
+    "package": "cp -r package.json package-lock.json README.md dist",
```
Comment (uittorio) [inline package.json:24]: I would like to understand a bit more the need of a package script. We are currently copying these files only for the release. If you were renaming package script to copyPackageAndReadmeToDist I would be okay.

Comment (martinjlowm) [reply package.json:24]: What about a scope similar to tests: `dist:package` and/or `dist:release` 🤔 ?

Comment (uittorio) [reply package.json:24]: I'm okay with dist:release :)

Comment (uittorio) [reply package.json]: @martinjlowm Sorry there was a misunderstanding. I think release can stay there. I would do write instead

```
"dist:release": "cp -r package.json package-lock.json README.md dist",
"release": "npm run dist:release && semantic-release",
```

Comment (martinjlowm) [reply package.json]: Morning :) Doneso!

Comment (uittorio) [reply package.json]: completely forgot about this. I'm okay with collect.. I'll merge it :)

**Pull request merged**